src/main/java/edu/ucar/unidata/rosetta/controller/TemplateController.java

```diff
@@ -64,29 +50,302 @@
 public class TemplateController implements HandlerExceptionResolver {
 
     protected static Logger logger = Logger.getLogger(TemplateController.class);
+
     @Autowired
     ServletContext servletContext;
+
+    @Resource(name = "dataManager")
+    private DataManager dataManager;
+    /*
+    @Resource(name = "jsonManager")
+    private JsonManager jsonManager;
+    */
     @Resource(name = "resourceManager")
     private ResourceManager resourceManager;
-    @Resource(name = "fileParserManager")
-    private FileParserManager fileParserManager;
+
+
+
+    /*
     @Resource(name = "netcdfFileManager")
     private NetcdfFileManager netcdfFileManager;
+
+    */
+    // Validators
+    @Autowired
+    private CFTypeValidator cfTypeValidator;
+
+    @InitBinder
+    public void initBinder(WebDataBinder binder) {
+        StringTrimmerEditor stringTrimmer = new StringTrimmerEditor(true);
+        binder.registerCustomEditor(String.class, stringTrimmer);
+        binder.setValidator(cfTypeValidator);
+    }
+
     @Resource(name = "fileValidator")
     private FileValidator fileValidator;
 
-    private String getDownloadDir() {
-        String downloadDir = "";
-        downloadDir = RosettaProperties.getDownloadDir();
-        return downloadDir;
+
+    /**
+     * Accepts a GET request for access to step 1 of the wizard.
+     *
+     * @param model  The Model object to be populated.
+     * @param request   HttpServletRequest needed to get the cookie.
+     * @return  View and the Model for the wizard to process.
+     */
+    @RequestMapping(value = "/step1", method = RequestMethod.GET)
+    public ModelAndView displayCFTypeSelectionForm(Model model, HttpServletRequest request) {
+
+        // Have we visited this page before during this session?
+        Cookie rosettaCookie = WebUtils.getCookie(request, "rosetta");
+
+        // Create a Data form-backing object.
+        Data data;
+        if (rosettaCookie != null)
+            // User-provided cfType info already exists.  Populate Data object with info.
+            data = dataManager.lookupById(rosettaCookie.getValue());
+        else
+            data = new Data();
+
+        // Add data object to Model.
+        model.addAttribute("data", data);
+        // Add current step to the Model (used by view to keep track of where we are in the wizard).
+        model.addAttribute("currentStep", "1");
+        // Add a list of all steps to the Model for rendering left nav menu.
+        model.addAttribute("steps", resourceManager.loadResources().get("steps"));
+        // Add domains data to Model (for platform display).
+        model.addAttribute("domains", resourceManager.loadResources().get("domains"));
+        // Add platforms data to Model (for platform selection).
+        model.addAttribute("platforms", resourceManager.loadResources().get("platforms"));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "platforms" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)